cpp-inference/inference_scaling_imagenet.cpp

```diff
@@ -74,64 +74,117 @@ void run_mnist(const std::string& model_name,
                const std::string& script_name,
                std::ofstream& timing_file)
 {
+  std::string context("Run Inference");
   int rank;
   MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+  log_data(context, LLDebug, "rank: " + std::to_string(rank) + " initiated");
 
   if (!rank)
+    log_data(context, LLInfo, "Connecting clients");
     std::cout<<"Connecting clients"<<std::endl<<std::flush;
 
   double constructor_start = MPI_Wtime();
   bool cluster = get_cluster_flag();
   bool is_colocated = get_colo();
-  SmartRedis::Client client(cluster);
+  SmartRedis::Client client(cluster, context);
   double constructor_end = MPI_Wtime();
   double delta_t = constructor_end - constructor_start;
   timing_file << rank << "," << "client()" << ","
               << delta_t << std::endl << std::flush;
+  //print rank # storing client() for debugging
+  std::string client_text = "client() time stored for rank: ";
+  client_text += std::to_string(rank);
+  log_data(context, LLDebug, client_text);
+
   std::string device = get_device();
   int num_devices = get_num_devices();
   bool use_multigpu = (0 == device.compare("GPU")) && num_devices > 1;
   bool should_set = get_set_flag();
   std::string model_key = "resnet_model";
   std::string script_key = "resnet_script";
+  log_data(context, LLDebug, "Running rank: " + std::to_string(rank) + "with vars should_set: " + std::to_string(should_set) + " - num_device: " + std::to_string(num_devices) + " - use_multigpu: " + std::to_string(use_multigpu) + " - is_coloated: " + std::to_string(is_colocated) + " - cluster: " + std::to_string(cluster));
```

Comment:
Overlong log message. Also, for this much logic, build the string outside the log_data() call